accepted/single-file/bundler.md

```diff
@@ -9,46 +9,45 @@ Ideally, the bundler should be:
+Using a special marker for recognizing bundle-header (instead of simply writing the header at the end of the file) enables compatibility with other post-processing tools (such as `signtool`) which require their own content to be at the end of the file.
 
-* The actual files to be published into the single file (including the managed app)
-* A bundle header containing:
-  * The bundler tool version 
-  * A bundle identifier: which is a *path-compatible* cryptographically strong name. 
-    * This identifier is used to distinguish bundles for different versions of the same app.
-    * This identifier is used as part of the bundle extraction mechanism as described in [this document](extract.md).
-    * Currently, a new bundle identifier is generated for each bundle transformation. 
-      In future, the bundle identifiers should be generated by hashing the contents of the bundle -- so that bundle transformation is deterministic.
-  * Offset of the bundle manifest 
+### Bundle Identifier
 
-* A bundle manifest that describes: 
-  * The location of embeded files (offset and size)
-  * The type of the embedded files:  MSIL assemblies, ready-to-run assemblies, configuration files (ex: `app.deps.json` `app.runtimeconfig.json`), or others. 
+The bundle identifier is a *path-compatible* cryptographically strong name. 
 
-Every `AppHost` has a static variable that identifies the location of the single-file header (if any). 
-By default, this value is zero, which indicates that the `AppHost` is not a bundle. 
-The bundler tool rewrites this value with the location of the bundle header. 
-This ensures that the bundle-header is not position constrained (ex: at the end of the file), so that tools like `signtool` can post-process the bundle file.
+* This identifier is used to distinguish bundles for different versions of the same app.
+* This identifier is used as part of the bundle extraction mechanism as described in [this document](extract.md).
+* A new bundle identifier is generated for each bundle transformation.  
 
-### Implementation
+## Bundle Transformation
 
-The bundler should ideally be located close to the core-host, since their implementation is closely related. Therefore, the bundler will be implemented in the `core-setup` repo. 
+Given a .net core app published with respect to a specific runtime, the bundler transforms the `AppHost` binary to a single-file bundle through the following actions:
+
+* Append the managed app and its dependencies as a binary-blob at the end of the `AppHost` executable. 
+* Write meta-data headers and manifest that help locate the contents of the bundle.
+* Set the bundle-indicator in the `AppHost` to the offset of the bundle-header.
+
+## Implementation
 
 The bundling tool is implemented as a library in the [Microsoft.NET.HostModel](https://www.nuget.org/packages/Microsoft.NET.HostModel/) package. This library is used by the SDK in order to publish a .net core app as a single-file. 
```

Comment:
I was looking at it just recently. Is there a reason for why we have this split into `Microsoft.NET.HostModel` and a task that lives in the SDK? It may be useful to write down the reason for the split here. (I was expecting to have one .dll with the msbuild task that does everything.)

Reply:
Non-exhaustive list of reasons:

1. msbuild tasks need to be multi-targeted as they must carry all their dependencies

2. cli uses hostmodel to generate apphosts for global tools without running a build

3. we can keep all the localized error messages in the sdk repo where we have loc infrastructure

Reply:
Also it was deemed desirable to have reading and writing code in the same repo to more easily evolve the format and unit test

Reply:
Thanks @nguerrera. Some of these reasons are documented in the [Bundler's readme](https://github.com/dotnet/runtime/tree/master/src/installer/managed/Microsoft.NET.HostModel/Bundle) file.  I'll add the details here too.

Reply:
I have updated the ReadMe file, now that a few repos have changed: 
https://github.com/dotnet/runtime/pull/32583